### ISSUE TITLE
Control Plane changes for Remote Extensions

### ISSUE THREAD

**Issue by awestover** [neondatabase/neon#4738]
In #4715 we try to connect to an S3 bucket for a new feature: [remote extensions](https://github.com/neondatabase/neon/blob/alek_targz/docs/rfcs/024-extension-loading.md?rgh-link-date=2023-07-14T14%3A52%3A35Z).

In the [regression test](https://github.com/neondatabase/neon/pull/4715/files#diff-e7185d1a4e578dbe30eae83521748d4ed52da53d106bbb9f78334e474cd977aa) we pass the remote storage configuration as a CLI ARG, something like 
```
-r {
	"bucket": "neon-dev-extensions-eu-central-1", 
	"region": "eu-central-1", 
	"endpoint": null,
	"prefix": "5174747474"
}
```
and this is what `compute_ctl` expects. 

This issue: 
- **Now we need to do this from `go` control plane.**
- Also, we need to make sure that `compute_ctl` gets the appropriate AWS credentials (probably as environment variables)
- Finally, we need to make the e2e tests test remote extensions
```[tasklist]
### Tasks
- [ ] https://github.com/neondatabase/neon/pull/4750
- [ ] neondatabase/cloud#5896
```

**Comment by awestover** [comment]
will be resolved by #4750

**Comment by awestover** [comment]
completed
